File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        hash_set = set(nums)
        max_len = 0
        for num in nums:
            if num - 1 in hash_set:
                continue
            curr_len = 1
            while num + 1 in hash_set:
                curr_len += 1
                num += 1
            max_len = max(max_len, curr_len)

        return max_len

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max_len = 0, slow = 0;
        HashSet<Character> set = new HashSet<>();
        for (int fast = 0; fast < s.length(); fast++) {
            while (set.contains(s.charAt(fast))) {
                set.remove(s.charAt(slow++));
            }
            set.add(s.charAt(fast));
            max_len = Math.max(max_len, set.size());
        }
        return max_len;
    }
}
